test-framework/src/main/java/org/keycloak/test/framework/page/WelcomePage.java

```diff
@@ -45,4 +55,17 @@ public void assertUserCreated() {
         Assertions.assertTrue(pageAlert.getText().contains("User created"));
     }
 
+    public boolean isPasswordSet() {
+        return !(driver.getPageSource().contains("Create a temporary administrative user") ||
+                driver.getPageSource().contains("You will need local access to create the temporary administrative user.") ||
+                driver.getPageSource().contains("you first create a temporary administrative user. Later, to harden security, create a new permanent administrative user"));
+    }
+
+    public void navigateToAdminConsole() {
+        driver.get("http://localhost:8080/admin/master/console");
+    }
+
+    public String getWelcomeMessage() {
+        return getTextFromElement(welcomeMessage);
```

Comment:
Not sure why a util and static imports are needed when you can just do:

```suggestion
        return welcomeMessage.getText();
```